src/main/java/org/springframework/integration/dsl/channel/DslWireTap.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2014 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.integration.dsl.channel;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.BeanFactoryAware;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.context.Lifecycle;
+import org.springframework.integration.channel.ChannelInterceptorAware;
+import org.springframework.integration.channel.interceptor.VetoCapableInterceptor;
+import org.springframework.integration.channel.interceptor.WireTap;
+import org.springframework.integration.core.MessageSelector;
+import org.springframework.integration.dsl.support.MessageChannelReference;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.core.DestinationResolutionException;
+import org.springframework.messaging.support.ChannelInterceptor;
+import org.springframework.util.Assert;
+
+/**
+ * @author Gary Russell
+ * @since 1.0.2
+ *
+ */
+public class DslWireTap implements ChannelInterceptor, BeanFactoryAware, VetoCapableInterceptor, InitializingBean,
```

Comment:
I make similar DSL-specific classes as `package protected` do not confuse end-users